andrei-s/src/main/java/org/java/mentorship/andrei_s/service/SongService.java

```diff
@@ -0,0 +1,74 @@
+package org.java.mentorship.andrei_s.service;
+
+import org.java.mentorship.andrei_s.domain.Song;
+import org.springframework.stereotype.Service;
+
+import java.util.LinkedList;
+
+@Service
+public class SongService {
+
+    private final LinkedList<Song> songs;
+    private int currentId = 0;
+
+    SongService()
+    {
+        this.songs = new LinkedList<>();
+
+        this.createNewSong(new Song(1, "Pop", 10, 10));
+        this.createNewSong(new Song(2, "Pop", 10, 10));
+        this.createNewSong(new Song(3, "Rock", 10, 11));
+        this.createNewSong(new Song(4, "Rock", 10, 11));
+        this.createNewSong(new Song(5, "Rock", 10, 12));
+    }
+
+    public Song getSongById(int songId)
+    {
+        for (Song song : songs)
+            if (song.getId() == songId) return song;
+
+        return null;
+    }
+
+    public LinkedList<Song> getSongs(String style)
+    {
+        LinkedList<Song> filteredSongs = new LinkedList<>();
+
+        for (Song song : this.songs)
+            if (song.getStyle().equals(style) || style == null) filteredSongs.add(song);
```

Comment:
Again, if you were to use Java 8 Streams, this would transform to

songs.stream().filter(song -> song.getStyle() == style).collect(Collectors.toList())